### PR TITLE
Upgrade to Undertow 2.2.4.Final

### PR DESCRIPTION
**Pull request by schlosna**
## Before this PR
Older Undertow is impacted by [CVE-2023-1108](https://access.redhat.com/security/cve/cve-2023-1108) with some JDKs.

See https://github.com/palantir/conjure-java-runtime/pull/2580

## After this PR
==COMMIT_MSG==
Upgrade to Undertow 2.2.4.Final
==COMMIT_MSG==

### CODE REVIEW

**Review by carterkozak — APPROVED** [review]

**Comment by carterkozak** [comment]
I've removed 'autorelease' since this dependency is only used in tests

File: versions.lock [hunk 1]
```diff
@@ -77,7 +77,7 @@ com.squareup.okio:okio:1.17.2 (1 constraints: 850cc309)
 commons-logging:commons-logging:1.2 (2 constraints: 8215ead1)
 de.erichseifert.vectorgraphics2d:VectorGraphics2D:0.13 (1 constraints: 8c0a80bb)
 de.rototor.pdfbox:graphics2d:0.25 (1 constraints: 8f0a84bb)
-io.undertow:undertow-core:2.2.18.Final (1 constraints: 57074361)
+io.undertow:undertow-core:2.2.24.Final (1 constraints: 54072f61)
```
Comment (schlosna) [inline versions.lock:80]: do not merge yet, will hold until com.palantir.conjure.java & com.palantir.conjure.java.runtime bumps

**Comment by schlosna** [comment]
once CJ & CJR are fully published, we can bump those if needed

**Comment by carterkozak** [comment]
sorry, I approved too quickly

**Pull request merged**